### PR TITLE
Heart Attack Syndrome

### PR DESCRIPTION
**Pull request by Shadowmech88**
Adds new stage 4 virus symptom, heart attack syndrome.
If the infected has a heart, this symptom causes the infected's heart to burst out of their chest and begin attacking them.
It can only trigger once, so you don't have to worry about being unable to replace their heart before curing the virus.

:cl:
 * rscadd: Added heart attack syndrome as a new stage 4 virus symptom, it causes your heart to burst out of your chest and begin attacking you.

### CODE REVIEW

**Comment by SkowronX** [comment]
>Stage 1

Holy shit

**Comment by ghost** [comment]
I honestly wasn't expecting this.

**Comment by TheFriggenSawbones** [comment]
you're the best coder right now

**Comment by carshalash** [comment]
SHEER HEART ATTACK

**Comment by ghost** [comment]
IT HAS NO WEAKNESS!

**Comment by Exxion** [comment]
>Stage 1

It had better be rare as fuck

**Comment by ghost** [comment]
@Exxion Stage one in the code is the same as stage 4 on a virus in game.

**Comment by Shadowmech88** [comment]
It's not, but doesn't the absence of a heart simply cause you to slowly take suffocation damage? That's what I remember.

**Comment by ghost** [comment]
I'd image that having no heart would be instant death 😕

**Comment by Exxion** [comment]
I'm pretty sure it causes you to very rapidly take suffocation damage
A few points of damage to your heart is enough to deal noticeable suffocation damage

But if Talvan is correct then never mind

**Comment by Shadowmech88** [comment]
Seems to have changed since I last checked, in testing the loss of a heart caused me to instantly pass out and quickly suffocate to death.

Stage 4 it is, then.

**Comment by ghost** [comment]
Pretty brutal.

File: code/modules/virus2/effect/effect.dm [hunk 1]
```diff
@@ -1455,6 +1430,37 @@ datum/disease2/effect/lubefoot/deactivate(var/mob/living/carbon/mob)
...
+			H.visible_message("<span class='danger'>\The [H]'s heart bursts out of \his chest!</span>","<span class='danger'>Your heart bursts out of your chest!</span>")
+			var/obj/item/organ/blown_heart = H.remove_internal_organ(H,H.get_heart(),H.get_organ(LIMB_CHEST))
+			var/list/spawn_turfs = list()
+			for(var/turf/T in orange(1, H))
+				if(!T.density)
+					spawn_turfs.Add(T)
+			if(!spawn_turfs.len)
+				spawn_turfs.Add(get_turf(H))
+			var/mob/living/simple_animal/hostile/heart_attack = new(pick(spawn_turfs))
+			heart_attack.name = blown_heart.name
```
Comment (Exxion) [inline code/modules/virus2/effect/effect.dm]: Any reason not to just do `heart_attack.appearance = blown_heart` then change the `desc`?

**Comment by gbasood** [comment]
@Shadowmech88 unless it's changed, heart being missing meant that it'd set your blood volume to 0, meaning passing out from lack of oxygen followed by oxydeath

**Pull request merged**